Version bump for edx-enterprise

brittneyexline opened the pull request.

brittneyexline commented on requirements/edx/base.in:
```diff
@@ -75,7 +75,7 @@ edx-django-release-util             # Release utils for the edx release pipeline
 edx-django-utils
 edx-drf-extensions
-edx-enterprise==1.6.9
+edx-enterprise
```
@jmbowman - i noticed you had pinned edx-enterprise here - any reason not to unpin?

tuchfarber commented:
In effect this would auto upgrade to the newest release of edx-enterprise. Is the goal here to make it so releasing a new version of edx-enterprise means it's ready to go out? I was never sure why we manually upgraded edx-enterprise in the past.

brittneyexline commented:
@tuchfarber that's a good question that i'm not sure we've been consistent with as a team. my opinion is that anything merged to master generally means that it should be releasable, and we should only pin when necessary and for temporary circumstances.

tuchfarber commented:
:+1: I fully agree! We still have "wait to release a new version" as a holdback in case we need master to wait.

tuchfarber commented:
I didn't see the `catalog` part of this PR when I first looked at it. Is that a change that came with `1.6.11`?

brittneyexline commented:
@tuchfarber - technically it came with 1.6.10 - we removed the catalog field from the enterprise customer object.  we had already removed code referencing the field, and it appears this test code is lingering... i don't see any other direct usages in this repo.

jmbowman commented:
I pinned it because tests initially failed when I tried upgrading it, due to the model change.  I'm open to edx-enterprise either being pinned by default or not, I'd just ask that if it isn't pinned, changes like this should be made much more carefully so they don't routinely trip up people outside the Enterprise team who are just trying to upgrade something else in edx-platform.  I recently added a segment about this in OEP-18:

> If a dependency is maintained by edX and only used in a few repositories, consider if it should stay pinned to a specific version to facilitate managing new releases. Best practice is to avoid making backwards-incompatible new releases whenever possible, but this can require excessive effort for a package only used in 1-2 repositories.

brittneyexline commented:
understood - i totally agree @jmbowman. this was definitely a miss by us in this case.

seeing some of the changes going on in the past few weeks, i think we need to create some stronger guidelines and best practices for engineers working in edx-enterprise, especially since most don't always have the context on decisions made by past engineers. this has resulted in multiple places of confusion recently with regards to how this package gets installed in the edx-platform runtime.

brittneyexline commented:
jenkins run bokchoy

muhammad-ammar commented:
@brittneyexline @jmbowman This is a mistake on my side. I should have tested the edx-enterprise changes with edx-platform prior to make a new edx-enterprise release. Actually I tried the `make upgrade` the dependencies in the platform after the edx-enterprise release but was unable to do so because of `edx-opaque-key` pkg issue. My thinking was that after a successful `make upgrade`, I will create a PR and then I will see if there are any issues caused by edx-enterprise new release but this didn't happened. Anyway I will be more careful in future.  

@brittneyexline Thanks a lot for taking care of this.

georgebabey commented:
@brittneyexline you'll want to run `make upgrade` again to pull in edx-enterprise version 1.6.12

brittneyexline commented:
no worries @muhammad-ammar - the way we use edx-enterprise makes it easy to make mistakes because you have to keep edx-platform and edx-enterprise in sync with each other on multiple dimensions that aren't readily apparent.

brittneyexline commented:
jenkins run bokchoy

fysheets commented:
jenkins run bokchoy

brittneyexline commented:
jenkins run bokchoy

bderusha commented:
Flakey Test.  Merging
